src/gt4py/next/embedded/nd_array_field.py

```diff
@@ -254,3 +253,48 @@ class JaxArrayField(_BaseNdArrayField):
         array_ns: ClassVar[ModuleType] = jnp
 
     common.field.register(jnp.ndarray, JaxArrayField.from_array)
+
+
+def _get_slices_with_named_indices(field: common.Field, named_indices: NamedIndicesOrRanges) -> tuple[slice | None, ...]:
+    """Generate slices based on named indices or ranges for a given field.
+
+        This function generates a tuple of slices, which can be used to extract sub-arrays from a field based on the
+        provided named indices or ranges. The named indices or ranges specify the dimensions and ranges of the sub-arrays
+        to be extracted.
+
+        Args:
+            field (common.Field): The field from which sub-arrays will be extracted.
+            named_indices (NamedIndicesOrRanges): A list of tuples containing dimension names and ranges.
+
+        Returns:
+            tuple[slice | None, ...]: A tuple of slices where each element corresponds to a slice along a specific dimension
+                                      of the field. If a dimension is not present in the named indices or ranges, a None
+                                      element is used to indicate expanding the dimension along that axis.
+
+        Note:
+            - The function handles cases where named indices or ranges are provided for dimensions that do not exist in the
+              field's domain.
+            - Slices are generated according to the specified indices or ranges, and can be used to extract sub-arrays using
+              indexing.
+        """
+    old_domain = field.domain
```

Comment:
my understanding is that this works for `len(field.domain) == len(named_indices)`, if that's the case let's add an assert.

Reply:
And then in an next step, let's think about the following:
```
f: Field[[I,J],...]
f[(I, UnitRange(...),)]
```
is this legal? should it be?

Reply:
Regarding this: 

```
f: Field[[I,J],...]
f[(I, UnitRange(...),)]
```

Currently I have implemented something similar as a test here: https://github.com/GridTools/gt4py/pull/1315/files#diff-7e004bcaf958ca74ad45bb6bfe6c0aaf90bc110f6c466923e5da97c54af16b20R160-R165

Currently it is legal to slice a field with a smaller domain and then it just returns that dimension and corresponding range, if it should not be legal we would have to add a check to check whether the named_indices are smaller than the `field.domain`